linetools/analysis/continuumfnd.py

```diff
@@ -0,0 +1,206 @@
+"""
+  Module for automatically finding continuum
+"""
+
+import numpy as np
+from matplotlib import pyplot as plt
+import linetools.utils as ltu
+
+from linetools.spectra.io import readspec
+import linetools.spectra.xspectrum1d as xspec
+import pdb
+
+
+def contknots(spec, sm=10, npix=40, lchmin=20, ewsnlim=2,
+              showcont=True, lam0=1500, dlam=20, yfmax=2, ymin=0,
+              outf=None, overwrite=False):
+    """ Finds continuum knots, and pixels that are located on continuum .
+        Identifies pixels that correspond to absorption lines, and do not includes them in continuum.
+
+        Parameters sm, npix, lchmin, ewsnlim that best define continuum might be different for different data sets.
+        Recomended using lt_continuumfit script (gui) after.
+
+    Parameters
+    ----------
+    spec : XSpectrum1D or str
+      input spectrum, or a file that contains input XSpectrum1D spectrum
+    sm : float
```

Comment:
if the parameter is option, indicate it as such.

Reply:
Indicated optional parameters